src/Compilers/CSharp/Test/Symbol/Symbols/Source/EventTests.cs

```diff
@@ -325,6 +327,119 @@ class C
+                Assert.Equal(1, e2.GetAttributes(AttributeDescription.DynamicAttribute).Count());
+                Assert.Equal(1, p.GetAttributes(AttributeDescription.DynamicAttribute).Count());
+            };
+            CompileAndVerify(source: source, additionalRefs: new[] { MscorlibRef, SystemCoreRef }, symbolValidator: validator);
+        }
+
+        [Fact]
+        public void EventFieldWithDynamicAttribute_CannotCompileWithoutSystemCore()
+        {
+            var source = @"
+public class A
+{
+    public event System.Action<dynamic> E1;
+}";
+            var libComp = CreateCompilationWithMscorlib(text: source).VerifyDiagnostics(
+                // (4,32): error CS1980: Cannot define a class or member that utilizes 'dynamic' because the compiler required type 'System.Runtime.CompilerServices.DynamicAttribute' cannot be found. Are you missing a reference?
+                //     public event System.Action<dynamic> E1;
+                Diagnostic(ErrorCode.ERR_DynamicAttributeMissing, "dynamic").WithArguments("System.Runtime.CompilerServices.DynamicAttribute").WithLocation(4, 32),
+                // (4,41): warning CS0067: The event 'A.E1' is never used
+                //     public event System.Action<dynamic> E1;
+                Diagnostic(ErrorCode.WRN_UnreferencedEvent, "E1").WithArguments("A.E1").WithLocation(4, 41));
+        }
+
+        [Fact]
+        public void EventWithDynamicAttribute_CannotCompileWithoutSystemCore()
+        {
+            var source = @"
+public class A
+{
+    public event System.Action<dynamic> E1 { add {} remove {} }
+}";
+            var libComp = CreateCompilationWithMscorlib(text: source).VerifyDiagnostics(
+                // (4,32): error CS1980: Cannot define a class or member that utilizes 'dynamic' because the compiler required type 'System.Runtime.CompilerServices.DynamicAttribute' cannot be found. Are you missing a reference?
+                //     public event System.Action<dynamic> E1 { add {} remove {} }
+                Diagnostic(ErrorCode.ERR_DynamicAttributeMissing, "dynamic").WithArguments("System.Runtime.CompilerServices.DynamicAttribute").WithLocation(4, 32));
+        }
+
+        [Fact]
+        public void EventWithDynamicAttribute_IsLoadedAsDynamic()
+        {
+            var libText = @"
+public class C {
+    public event System.Action<dynamic> E1;
+}";
+            var libComp = CreateCompilationWithMscorlib(text: libText, references: new[] { MscorlibRef, SystemCoreRef });
+            Assert.False(libComp.GetDiagnostics().Any(d => d.Severity == DiagnosticSeverity.Error));
+
+            byte[] libImage;
+            using (var output = new MemoryStream())
+            {
+                libComp.Emit(output, options: new EmitOptions(metadataOnly: true));
+                libImage = output.ToArray();
+            }
+            var libAssemblyRef = MetadataReference.CreateFromImage(libImage.AsImmutableOrNull());
```

Comment:
Thanks. Will do.

From: Charles Stoner [mailto:notifications@github.com]
Sent: Saturday, February 27, 2016 12:22 AM
To: dotnet/roslyn <roslyn@noreply.github.com>
Cc: Julien Couvreur <jcouv@microsoft.com>
Subject: Re: [roslyn] Fix for 7845: add DynamicAttribute to event with dynamic type (#9266)


In src/Compilers/CSharp/Test/Symbol/Symbols/Source/EventTests.cs<https://github.com/dotnet/roslyn/pull/9266#discussion_r54330430>:

> +        public void EventWithDynamicAttribute_IsLoadedAsDynamic()

> +        {

> +            var libText = @"

> +public class C {

> +    public event System.Action<dynamic> E1;

> +}";

> +            var libComp = CreateCompilationWithMscorlib(text: libText, references: new[] { MscorlibRef, SystemCoreRef });

> +            Assert.False(libComp.GetDiagnostics().Any(d => d.Severity == DiagnosticSeverity.Error));

> +

> +            byte[] libImage;

> +            using (var output = new MemoryStream())

> +            {

> +                libComp.Emit(output, options: new EmitOptions(metadataOnly: true));

> +                libImage = output.ToArray();

> +            }

> +            var libAssemblyRef = MetadataReference.CreateFromImage(libImage.AsImmutableOrNull());

using block can be replaced with var libAssemblyRef = libComp.EmitToImageReference();.

—
Reply to this email directly or view it on GitHub<https://github.com/dotnet/roslyn/pull/9266/files#r54330430>.